### DISCUSSION TITLE
v0.67.0

### DISCUSSION THREAD

**Discussion by tommoor** [outline/outline#4681] [Announcements]
### Breaking changes

> **Warning** As of this release all documents edits are sent over websockets, this results in a much better editor experience with full support for realtime collaborative editing but _requires_ websockets. If your self-hosted setup was managing without working websocket connections editing will no longer work with this release.

> **Warning** Any Redis username in the `REDIS_URL` environment variable will be passed through to Redis as of this release, they were previously ignored. If you see NOAUTH errors then remove the username from the url config.

> **Warning** Detection for HTTPS connections to proxies changed. If your proxy terminates SSL and the installation is in a https redirect loop after updating ensure that your proxy is adding the `x-forwarded-proto: https` header.

### Enhancements

* Add HTML export option in https://github.com/outline/outline/pull/4056
* Images now have a full width display option in https://github.com/outline/outline/pull/4389
* Image resizing in https://github.com/outline/outline/pull/4368
* Added support for LaTeX math blocks in https://github.com/outline/outline/pull/4446
* Added Google Analytics 4 integration in https://github.com/outline/outline/pull/4626
* Added support for "phrase matching" and -negative keywords when searching
* Throttle email notifications upon updating document frequently by @apoorv-mishra in https://github.com/outline/outline/pull/4026
* New document insights panel in https://github.com/outline/outline/pull/4418
* Shared documents can now have a custom path by @apoorv-mishra in https://github.com/outline/outline/pull/4550
* Show diff when navigating revision history in https://github.com/outline/outline/pull/4069
* Updated collection header in https://github.com/outline/outline/pull/4101
* Add Dutch translations to language selector in https://github.com/outline/outline/pull/4120
* Add `SENTRY_TUNNEL` option by @K3das in https://github.com/outline/outline/pull/4298
* Option for separate edit mode in https://github.com/outline/outline/pull/4203
* Sign webhook requests by @apoorv-mishra in https://github.com/outline/outline/pull/4156
* Render diffs in email notifications in https://github.com/outline/outline/pull/4164
* Introduce account preferences to remember user's previous location by @apoorv-mishra in https://github.com/outline/outline/pull/4126
* Updated designs for pinned docs in https://github.com/outline/outline/pull/4124
* Added syntax highlighting for various languages by @wale in https://github.com/outline/outline/pull/4341
* Code blocks can now optionally display line numbers in https://github.com/outline/outline/pull/4324
* Drafts can now be moved between collections in https://github.com/outline/outline/pull/4652
* New option to search titles only in https://github.com/outline/outline/pull/4587
* User avatars displayed in author dropdown in https://github.com/outline/outline/pull/4551
* Hundreds of updated translations

### Fixes

* Animation of InputSelect is janky in https://github.com/outline/outline/pull/4061
* Improving the `urls` to not break protocols by @iifawzi in https://github.com/outline/outline/pull/3995
* Dialog doesn't close after deleting a document with a parent by @vgwidt in https://github.com/outline/outline/pull/4108
* Enter in table cell adds a row after the current one by @PabloYG in https://github.com/outline/outline/pull/4186
* Share document link that opens full editor by @dolsem in https://github.com/outline/outline/pull/4134
* Add cursor style user preference in https://github.com/outline/outline/pull/4199
* Remove invite button should not overlap with member dropdown button by @apoorv-mishra in https://github.com/outline/outline/pull/4243
* Multiplayer changes potentially attributed to incorrect user in https://github.com/outline/outline/pull/4282
* Set subscribe/unsubscribe state correctly for documents by @apoorv-mishra in https://github.com/outline/outline/pull/4266
* Disallow adding self to collection in https://github.com/outline/outline/pull/4299
* perf: Don't go to disk for html more than once in https://github.com/outline/outline/pull/4312
* chore: Remove method override middleware in https://github.com/outline/outline/pull/4315
* Authentication provider display in https://github.com/outline/outline/pull/4332
* Return correct permissions upon document update by @apoorv-mishra in https://github.com/outline/outline/pull/4352
* Allow viewers to upload avatar in https://github.com/outline/outline/pull/4349
* Allow viewers to upload into documents where they have edit permission for the collection in https://github.com/outline/outline/pull/4468
* Unable to login with matching email from another auth provider in https://github.com/outline/outline/pull/4356
* perf: Avoid fsstat on every request, remove `koa-static` in https://github.com/outline/outline/pull/4387
* Removed usage of tiley in https://github.com/outline/outline/pull/4406
* `DATABASE_URL` missing in env does not produce validation error in https://github.com/outline/outline/pull/4409
* Self-hosted logic for allowed domains in https://github.com/outline/outline/pull/4412
* Images and files with cyrillic names now import correctly in https://github.com/outline/outline/pull/4654
* Lists inside of tasks are now styled correctly in https://github.com/outline/outline/pull/4648
* Links between documents are now correct in exports https://github.com/outline/outline/pull/4639
* Fixed extra space rendered around soft breaks in https://github.com/outline/outline/issues/4616
* Added a guard to ensure name is returned from OIDC providers in https://github.com/outline/outline/issues/4453
* Uploaded and immediately deleted attachments are now correctly removed from storage in https://github.com/outline/outline/pull/4562
* Improved handling of pasting into document title in https://github.com/outline/outline/pull/4474

## New Contributors
* @iifawzi made their first contribution in https://github.com/outline/outline/pull/3995
* @vgwidt made their first contribution in https://github.com/outline/outline/pull/4108
* @PabloYG made their first contribution in https://github.com/outline/outline/pull/4186
* @dolsem made their first contribution in https://github.com/outline/outline/pull/4134
* @kgeorgiou made their first contribution in https://github.com/outline/outline/pull/4211
* @pbkompasz made their first contribution in https://github.com/outline/outline/pull/4216
* @chavda-bhavik made their first contribution in https://github.com/outline/outline/pull/4247
* @mastqe made their first contribution in https://github.com/outline/outline/pull/4259
* @K3das made their first contribution in https://github.com/outline/outline/pull/4298
* @Daniyaalbeg made their first contribution in https://github.com/outline/outline/pull/4323
* @MihirGH made their first contribution in https://github.com/outline/outline/pull/4336
* @wale made their first contribution in https://github.com/outline/outline/pull/4341
* @ItaloSa made their first contribution in https://github.com/outline/outline/pull/4365

**Full Changelog**: https://github.com/outline/outline/compare/v0.66.0...v0.67.0

<hr /><em>This discussion was created from the release <a href='https://github.com/outline/outline/releases/tag/v0.67.1'>v0.67.1</a>.</em>

**Comment by PhilReinking** [comment]
@tommoor after updating my self-hosted instance to `0.67.1` I get the following errors when trying to view documents:

Do you have any idea what could cause that? When I roll back to `0.66.3`, it works again.

```
TypeError: Cannot destructure property 'id' of 'ctx.body' as it is undefined.
    at ./build/server/routes/api/documents.js:393:5
    at ./node_modules/dd-trace/packages/datadog-instrumentations/src/koa.js:88:57
    at ./node_modules/dd-trace/packages/datadog-shimmer/src/shimmer.js:22:21
    at dispatch (./node_modules/koa-router/node_modules/koa-compose/index.js:44:32)
    at next (./node_modules/koa-router/node_modules/koa-compose/index.js:45:18)
    at authMiddleware (./build/server/middlewares/authentication.js:141:12)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async apiWrapperMiddleware (./build/server/routes/api/middlewares/apiWrapper.js:14:5)
    at async userAgent (./node_modules/koa-useragent/dist/index.js:12:5)
    at async ./node_modules/koa-mount/index.js:58:5
    at async ./node_modules/koa-mount/index.js:52:26 {
  status: 500
}
TypeError: Cannot destructure property 'id' of 'ctx.body' as it is undefined.
    at ./build/server/routes/api/documents.js:393:5
    at ./node_modules/dd-trace/packages/datadog-instrumentations/src/koa.js:88:57
    at ./node_modules/dd-trace/packages/datadog-shimmer/src/shimmer.js:22:21
    at dispatch (./node_modules/koa-router/node_modules/koa-compose/index.js:44:32)
    at next (./node_modules/koa-router/node_modules/koa-compose/index.js:45:18)
    at authMiddleware (./build/server/middlewares/authentication.js:141:12)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async apiWrapperMiddleware (./build/server/routes/api/middlewares/apiWrapper.js:14:5)
    at async userAgent (./node_modules/koa-useragent/dist/index.js:12:5)
    at async ./node_modules/koa-mount/index.js:58:5
    at async ./node_modules/koa-mount/index.js:52:26 {
  status: 500
}
```

**Reply by tommoor** [reply]
Remove your `build` folder and re-run `yarn build` – the build folder contains old code from the last release that is conflicting.

**Reply by PhilReinking** [reply]
Thanks! That solved it.

Maybe adding the `clean` command to the `heroku-postbuild`?
